File: src/main/java/NekoEssentials/commands/vanish.java
package NekoEssentials.commands;

import org.bukkit.Bukkit;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.bukkit.potion.PotionEffect;
import org.bukkit.potion.PotionEffectType;

import static NekoEssentials.Main.c;
import static NekoEssentials.Main.config;

public class vanish implements CommandExecutor {

    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
        if (label.equalsIgnoreCase("vanish")) {
            if (sender instanceof Player) {
                Player player = (Player) sender;
                if (player.hasPermission("NekoEssentials.vanish")) {
                    if (player.hasPotionEffect(PotionEffectType.INVISIBILITY)) {
                        player.removePotionEffect(PotionEffectType.INVISIBILITY);
                        for (Player onlinePlayer : Bukkit.getOnlinePlayers()) {
                            onlinePlayer.showPlayer(player);
                        }
                        player.sendMessage(c(config.getConfig().getString("vanish-off")));
                    } else {
                        player.addPotionEffect(new PotionEffect(PotionEffectType.INVISIBILITY, Integer.MAX_VALUE, 0, false, false));
                        for (Player onlinePlayer : Bukkit.getOnlinePlayers()) {
                            onlinePlayer.hidePlayer(player);
                        }
                        player.sendMessage(c(config.getConfig().getString("vanish-on")));
                    }
                    return true;
                } else {
                    player.sendMessage(c(config.getConfig().getString("no-permission")));
                }
            }
        }
        return false;
    }
}

File: src/main/java/nekoutility/commands/vanish.kt
package nekoutility.commands

import nekoutility.Main.Companion.send
import org.bukkit.Bukkit
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class vanish : CommandExecutor {
    private val vanishedPlayers: MutableSet<Player> = HashSet()
    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {
            if (sender is Player) {
                val player = sender
                return if (player.hasPermission("nekoutility.vanish")) {
                    if (isVanished(player)) {
                        for (onlinePlayer in Bukkit.getOnlinePlayers()) {
                            onlinePlayer.showPlayer(player)
                        }
                        vanishedPlayers.remove(player)
                        send(player, "vanish-off")
                    } else {
                        for (onlinePlayer in Bukkit.getOnlinePlayers()) {
                            onlinePlayer.hidePlayer(player)
                        }
                        vanishedPlayers.add(player)
                        send(player,"vanish-on")
                    }
                    true
                } else {
                    send(player,"no-permission")
                    true
                }
            }
        return false
    }

    private fun isVanished(player: Player): Boolean {
        return vanishedPlayers.contains(player)
    }
}